core/consensus/component.go

```diff
@@ -301,9 +299,20 @@ func (c *Component) propose(ctx context.Context, duty core.Duty, value proto.Mes
 		Receive:   t.recvBuffer,
 	}
 
+	// Instrument consensus instance.
+	t0 := time.Now()
+	def := c.def
+	def.Decide = func(ctx context.Context, duty core.Duty, val [32]byte, qcommit []qbft.Msg[core.Duty, [32]byte]) {
+		instrumentConsensus(duty, qcommit[0].Round(), t0)
+		c.def.Decide(ctx, duty, val, qcommit)
+	}
+
 	// Run the algo, blocking until the context is cancelled.
-	err = qbft.Run[core.Duty, [32]byte](ctx, c.def, qt, duty, peerIdx, hash)
-	if err != nil && !isContextErr(err) {
+	err = qbft.Run[core.Duty, [32]byte](ctx, def, qt, duty, peerIdx, hash)
+	if isContextErr(err) {
+		consensusTimeout.WithLabelValues(duty.String()).Inc()
```

Comment:
think about this one some more

Reply:
It times out when context is closed but we haven't DECIDED